source/fundamentals/connection/tls.txt

```diff
@@ -152,6 +155,36 @@ please refer to the `JSSE Reference Guide
 
 .. _tls-disable-hostname-verification:
 
+Configure a Client-Specific Trust Store and Key Store
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can configure a client-specific trust store and key store using the
+``init()`` method of the ``SSLContext`` class.
+
+The following code snippet configures a client-specific trust store and key
+store:
+
+.. code-block:: java
+
+   String algorithm = "TLS";
+   try {
+      final SSLContext sslContext = SSLContext.getInstance(algorithm);
+      sslContext.init(<Your Key Managers>, <Your Trust Managers>, <A Secure Random Number Generator>);
+      MongoClientSettings settings = MongoClientSettings.builder()
+               .applyToSslSettings(builder -> {
+                     builder.enabled(true);
+                     builder.context(sslContext);
+               })
+               .build();
+      MongoClient client = MongoClients.create(settings);
+   }
+   catch (NoSuchAlgorithmException e) {
+      System.err.println(algorithm + " is not a valid message digest algorithm");
+   }
+
+For more information on the ``SSLContext`` class, see the official Oracle API
+documentation for `SSL Context <https://docs.oracle.com/en/java/javase/16/docs/api/java.base/javax/net/ssl/SSLContext.html>`__. 
```

Comment:
> the official Oracle API documentation

`javax.net.ssl.SSLContext` is a part of the Java SE, and this specific URI simply points to a page that contains a part of the Java SE API documentation published by Oracle, i.e., it is not specific to Oracle JDK, which means there is no need to emphasize "Oracle API", as it is not Oracle-specific API.

Reply:
Thank you for the note. I have updated the API docs to the non-vendor specific link provided above and removed the reference to Oracle. Here is the fix: https://github.com/mongodb/docs-java/pull/139/commits/c2803079ee4485e005c4627f6b3f3da68cfa05f9

Apologies for my poor understanding of the Java ecosystem. For my own edification, do you think it is a good practice to link to OpenJDK API docs over Oracle JDK API Docs for Java SE libraries? Sorry if this is a poorly worded or silly question.

Reply:
No worries. Here are the ideas in a nutshell:

- if we are linking to a piece of documentation that is part of the Java SE, we should not imply that it is vendor-specific by saying "Oracle documentation" even if the linked Java SE documentation is published by Oracle or a different vendor;
- If we are linking to a piece of documentation that is not part of the Java SE and is vendor-specific, then we should point that out.

As for the [API docs published at docs.oracle.com](https://docs.oracle.com/en/java/javase/16/docs/api/index.html), one can differentiate between the Java SE API and Oracle JDK API by looking at the "Java SE" and "JDK"/"Other modules" tabs. Everything in modules listed under the "Java SE" tab is part of the Java SE, anything listed under "JDK"/"Other modules" is not.

> do you think it is a good practice in general to link to OpenJDK API docs over Oracle JDK API Docs for Java SE libraries

Since Java driver docs have been referring to docs published at docs.oracle.com, I think we should continue doing so. I provided the official Java SE API link (it is non-vendor-specific) just to support my words that "`javax.net.ssl.SSLContext` is a part of the Java SE`, not to suggest that we should start linking to a different place. Sorry if that caused confusion.

Reply:
Sorry for misinterpreting your comment, and thanks for all of the helpful background information! I will make sure to check what is part of the Java SE vs. a vendor specific release and use the correct language going forward. Fix: https://github.com/mongodb/docs-java/pull/139/commits/a6cbdf0922cfd5cd8e9664435988e653ac70ed0f